[Green-Tea-123] iP

Green-Tea-123 opened the pull request:
### **Rizzler**
> "Sticking out your gyat for the Rizzler" - [@ovp.9](https://www.tiktok.com/@ovp.9)

**Rizzler** is  an interactive chatbot that remembers _all_ your tasks so you don't have to.

**Rizzler** is:

- text-based
- supports a multitude of functions
- just unfathomably based

All you gotta do to have it is,

1. download it
2. run it
3. add your tasks
4. stick your gyat out for it 🥵

Features:

- [x] Task management
- [x] Task finding
- [ ] Imparting rizz (coming soon)

Are you a java programmer? Wanna take a peek at **Rizzler's** inner workings?
Here's its `main` method:
```java
public class Rizzler {
    public static void main(String[] args) {
        new Rizzler("data/saveData.txt").run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Green-Tea-123 committed:
Renamed duke to rizzler

Green-Tea-123 committed:
Added ability for rizzler to echo inputs

Green-Tea-123 committed:
Added ability to store and list tasks

Green-Tea-123 committed:
Added marking and unmarking of task completion

Green-Tea-123 committed:
Added support for different task types, together with some misinput handling

Green-Tea-123 committed:
Updated input and expected results for automated testing

Green-Tea-123 committed:
Added input handling and feedbck for all commands to rizzler

Green-Tea-123 committed:
Added function to delete tasks from the list, also added test cases for delete function and misinput handling.

Green-Tea-123 committed:
Add ability to save and load tasklist

Green-Tea-123 committed:
Merge branch 'branch-Level-7'

Green-Tea-123 committed:
Add date comprehension to rizzler

Green-Tea-123 committed:
Merge branch 'branch-Level-8'

Green-Tea-123 committed:
Configure rizzler to be more object-oriented

Green-Tea-123 committed:
Merge branch 'branch-A-MoreOOP'

Green-Tea-123 committed:
Put classes into package rizzler

Green-Tea-123 committed:
Merge branch 'add-gradle-support' of https://github.com/Green-Tea-123/ip

Green-Tea-123 committed:
Add JUnit test cases for TaskList and AddCommand

Green-Tea-123 committed:
Update build.gradle to support shadowJar

Green-Tea-123 committed:
Add JavaDocs to all of the classes in rizzler

Green-Tea-123 committed:
Merge branch 'branch-A-JavaDoc'

Green-Tea-123 committed:
Enforce coding standard to all classes

Green-Tea-123 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/rizzler/AddCommand.java
#	src/main/java/rizzler/Command.java
#	src/main/java/rizzler/DeleteCommand.java
#	src/main/java/rizzler/ExitCommand.java
#	src/main/java/rizzler/ListCommand.java
#	src/main/java/rizzler/MarkCommand.java
#	src/main/java/rizzler/UnmarkCommand.java

Green-Tea-123 committed:
Add ability to find tasks with a keyword

Green-Tea-123 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/rizzler/Task.java
#	src/main/java/rizzler/Ui.java

Green-Tea-123 committed:
Configure Rizzler to have a GUI with javafx

MAOXIONGKAI reviewed (commented):
LGTM, just a few stylings to improve on for better readability of the code in the future. Good luck on the project, thank you for your wonderful code and implementations

MAOXIONGKAI commented on src/main/java/rizzler/AddCommand.java:
```diff
@@ -0,0 +1,149 @@
+                String output = tasks.add(new Todo(taskName));
+                fileStorage.save(tasks.getListToSave());
+                return output;
+            }
```
For switch statement, if a case intended to fall through the next case(no break statement at the end), then you should include //Fallthrough comment at the end for clarity, else you should indicate clearly there's no fall through using break statement. Refer to: [switch statement coding standard](https://se-education.org/guides/conventions/java/intermediate.html)

MAOXIONGKAI commented on src/main/java/rizzler/AddCommand.java:
```diff
@@ -0,0 +1,149 @@
+            }
+            String fromTime = this.fullCommand[indexOfFrom + 1];
+            String toTime = this.fullCommand[indexOfTo + 1];
+            try {
```
Would it be better to add line break between different blocks of code for clarity? I feel like at least the variable declaration and other logics should be separated by line breaks.

MAOXIONGKAI commented on src/test/java/rizzler/AddCommandTest.java:
```diff
@@ -0,0 +1,88 @@
+                    + "deadline [task name] /by [yyyy-mm-dd]");
+        }
+        // deadline date format should not be wrong
+        try {
```
I feel like the code and the comments are very compact and the readability is compromised to a certain extend, would it be better to add line breaks for better clarity?🤔

MAOXIONGKAI commented on src/test/java/rizzler/TaskListTest.java:
```diff
@@ -0,0 +1,74 @@
+    @AfterEach
+    public void restoreStreams() {
+        System.setOut(originalOut);
+    }
```
I like the way you set up and clean up the test using **BeforeEach** and **AfterEach** annotations. Implementing advance knowledge from the course takes effort and you are doing it very well👍

MAOXIONGKAI reviewed (commented):
LGTM, just a few stylings to improve on for better readability of the code in the future. Good luck on the project, thank you for your wonderful code and implementations! 👍

tata32000 commented on src/main/java/rizzler/Deadline.java:
```diff
@@ -0,0 +1,68 @@
+package rizzler;
```
Maybe it's better to put the Task to its own package? For example, rizzler.task package

tata32000 commented on src/main/java/rizzler/DeleteCommand.java:
```diff
@@ -0,0 +1,64 @@
+     *
+     * @return <code>false</code> as command does not exit Rizzler.
+     */
+    public boolean isExit() {
```
Do you think isDone is a better name?

tata32000 commented on src/main/java/rizzler/FileStorage.java:
```diff
@@ -0,0 +1,114 @@
+     *
+     * @return ArrayList of <code>Task</code>.
+     */
+    ArrayList<Task> load() {
```
Should this be named loadTasks instead?

tata32000 reviewed (commented):
Maybe you can put Task and other class into a separate package like rizzler.task or rizzler.ui?

Green-Tea-123 committed:
Add assertion at the end of FileStorage creation.

Currently, there are no assertions in the code of
Rizzler, which may lead to assumptions in the code
that turn out to be wrong.

Adding assertions to the code eliminates such
assumptions and improves code quality.

As a step towards including assertions into the
code, let's introduce assertions into the
FileStorage class to assert save file creation.

Green-Tea-123 committed:
Merge pull request #1 from Green-Tea-123/branch-A-Assertions

Add assertion at the end of FileStorage creation

Green-Tea-123 committed:
Merge branch 'master' of https://github.com/Green-Tea-123/ip

* 'master' of https://github.com/Green-Tea-123/ip:
  Add assertion at the end of FileStorage creation.

Green-Tea-123 committed:
Extract methods in AddCommand for readability

The codebase has some methods that are considerably lengthy,
which inhibits readability and by extension code quality.

Extracting portions of those methods out into new methods
reduces the length of each method, increasing the readability
of the codebase.

Let's increase the readability of the codebase by extracting
the input checking of AddCommand during command execution into separate
methods to reduce the length of the execute() function.

Green-Tea-123 committed:
Merge pull request #2 from Green-Tea-123/branch-A-CodeQuality

Extract methods in AddCommand for readability

Green-Tea-123 committed:
Merge branch 'master' of https://github.com/Green-Tea-123/ip

* 'master' of https://github.com/Green-Tea-123/ip:
  Extract methods in AddCommand for readability

Green-Tea-123 committed:
Set up continuous integration

Green-Tea-123 committed:
Add function to sort tasks in Rizzler

Green-Tea-123 committed:
Merge pull request #3 from Green-Tea-123/branch-C-Sort

Add function to sort tasks in Rizzler

Green-Tea-123 committed:
Improve the UI of Rizzler

Green-Tea-123 committed:
Add more personality to Rizzler

Green-Tea-123 committed:
Add JUnit tests for more classes

Green-Tea-123 committed:
Update README for release
